src/race-details.ts

```diff
@@ -0,0 +1,20 @@
+import type { IKartRace } from "./models/go-kart.d.ts";
+
+export default function showRaceDetails(data: IKartRace) {
+  const raceDetailsElement = document.querySelector('#race-details');
+  try {
+    if (!raceDetailsElement) throw new Error('Race Details Element Not Found');
+    if (!data) throw new Error('No Data Provided');
+  } catch (error) {
```

Comment:
Do the above without the `try` and instead of throwing, just error out like the catch would have done in any case

Reply:
If the catch is on the caller of this function, then you can still throw